pkg/pgmodel/model/labels.go

```diff
@@ -164,18 +194,24 @@ func (l *Labels) Equal(b *Labels) bool {
 	return l.str == b.str
 }
 
-// Labels implements sort.Interface
-var _ sort.Interface = (*Labels)(nil)
-
-func (l *Labels) Len() int {
-	return len(l.Names)
+func (l *Labels) IsSeriesIDSet() bool {
+	return l.seriesID != unsetSeriesID
 }
 
-func (l *Labels) Less(i, j int) bool {
-	return l.Names[i] < l.Names[j]
+func (l *Labels) GetSeriesID() (SeriesID, SeriesEpoch, error) {
+	switch l.seriesID {
+	case unsetSeriesID:
+		return 0, 0, fmt.Errorf("Series id not set")
+	case 0:
```

Comment:
Is this possible? (for sake of knowing more)

Reply:
Not unless it's unset(programming error)

Reply:
If that has to be programming error, then we should have a explicit check for this in tests, so that it can be caught before merging.